.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,33 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main, release-2.2, release-1.4 ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main, release-2.2, release-1.4 ]
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
```

Comment:
Nit: since this repo is only Go, no real need to matrix the build, we could have just put it in the Initialize step, and since it's only a single language with no autobuild, we can fail-fast. But no reason to block the the merge.

Reply:
should `with.languages` be an array or a single string ?

Reply:
Looks like you got it sorted, I should have been clearer, but you can remove the entire `strategy` stanza as `fail-fast: true` is the default behavior. No need to change it we can address it another PR